src/domain/PrizeCalculator.js

```diff
@@ -0,0 +1,43 @@
+const { PRIZE_MATCH_NUMBER_COUNT } = require("../constants/condition.js");
+
+class PrizeCalculator {
+  calculatePrize(eachLottoNumbers, winningNumbers, bonusNumber) {
+    const eachCompareResult = eachLottoNumbers.map((lottoNumbers) => {
+      return this.getCompareResult(lottoNumbers, winningNumbers, bonusNumber);
+    });
+
+    const eachLottoPrize = eachCompareResult.map(this.getLottoPrize);
+
+    return eachLottoPrize;
+  }
+
+  getCompareResult(lottoNumbers, winningNumbers, bonusNumber) {
+    const matchedLottoNumberCount = this.getMatchedLottoNumberCount(lottoNumbers, winningNumbers);
+    const hasBonusNumber = this.hasBonusNumber(lottoNumbers, bonusNumber);
+
+    return { matchedLottoNumberCount, hasBonusNumber };
+  }
+
+  getMatchedLottoNumberCount(lottoNumbers, winningNumbers) {
+    return lottoNumbers.filter((number) => winningNumbers.includes(number)).length;
+  }
```

Comment:
reduce 메서드를 사용하면 바로 카운트된 결과를 반환해서 좋은 것 같아요 :)
```suggestion
  getMatchedLottoNumberCount(lottoNumbers, winningNumbers) {
    return lottoNumbers.reduce((matchCount, number) => {
      if(winningNumbers.includes(number)) matchCount += 1;
      return matchCount;
    }, 0);
  }
```

Reply:
reduce를 이렇게 사용하는것은 생각하지 못했네요 ! 좋은 방법 인것 같습니다.